www/utils/supabase/queries.ts

```diff
@@ -1,5 +1,41 @@
 import { cache } from 'react';
 import { SupabaseClient } from '@supabase/supabase-js';
+import { unstable_cache as next_unstable_cache } from 'next/cache';
+import { getFreeMessageCount } from './actions';
+
+export const unstable_cache = <Args extends any[], Output>(
+  callback: (...args: Args) => Promise<Output>,
+  key: string[],
+  options: { revalidate: number },
+) => {
+  return cache(
+    next_unstable_cache(
+      callback as unknown as (...args: any[]) => Promise<Output>,
+      key,
+      options
+    )
+  );
+};
+
+export const getChatAccess = unstable_cache(
+  async (supabase: SupabaseClient, userId: string) => {
+    const { data: subscription } = await supabase
+      .from('subscriptions')
+      .select('status, trial_end, metadata')
+      .in('status', ['trialing', 'active'])
+      .maybeSingle();
+    
+    const isSubscribed = !!(subscription && subscription.status === 'active' && !subscription.trial_end);
+    const freeMessages = isSubscribed ? 0 : (subscription?.metadata as { freeMessages: number })?.freeMessages ?? 0;    
+    return {
+      isSubscribed,
+      freeMessages,
+      canChat: isSubscribed || freeMessages > 0
+    };
+  },
+  ['chat-access'],
+  { revalidate: 60 }
+);
```

Comment:
Originally the getFreeMessageCount function would create a new trial subscription if one didn't exist, but by using this that call is never made and a trial subscription will never get created

Reply:
Either call that if no subscription is made or refactor the logic to use that function instead